File: year_2022/day_01/day01.go
package main

import (
	"aoc/tools"
	"fmt"
	"sort"
	"strconv"
)

func solve(input string) (int, int) {
	lines := tools.ReadLines(input)
	blocks := tools.SplitBlocks(lines)
	elves := make([]int, len(blocks))

	for i, elf := range blocks {
		value := 0

		for _, item := range elf {
			v, _ := strconv.Atoi(item)
			value += v
		}

		elves[i] = value
	}

	sort.Ints(elves)
	top3 := elves[len(elves)-3:]
	part1 := top3[2]
	part2 := 0

	for i := 0; i < 3; i++ {
		part2 += top3[i]
	}

	return part1, part2
}

func main() {
	fmt.Println(solve("test.txt"))
	fmt.Println(solve(tools.GetData(2022, 01)))
}

File: year_2022/day_01/day01.py
from tools import loader, parsers


def puzzle(data: list[list[str]]) -> tuple[int, int]:
    """
    >>> puzzle(parsers.blocks('test.txt'))
    (24000, 45000)
    """
    sums = sorted(sum(int(i) for i in block) for block in data)
    return sums[-1], sum(sums[-3:])


print(puzzle(parsers.blocks(loader.get())))  # 69883, 207576
